tests/api.c

```diff
@@ -32067,6 +32067,217 @@ tgZl96bcAGdru8OpQYP7x/rI4h5+rwA/kwIBAg==\n\
 #endif
 }
 
+/* test_EVP_Cipher_extra, Extra-test on EVP_CipherUpdate/Final. see also test.c */
+#if (defined(OPENSSL_EXTRA) || defined(OPENSSL_ALL)) &&\
+    (!defined(NO_AES) && defined(HAVE_AES_CBC) && defined(WOLFSSL_AES_128)) 
+static void binary_dump(void *ptr, int size)
+{
+    #ifdef WOLFSSL_EVP_PRINT
+   	int i = 0;
+	unsigned char *p = (unsigned char *) ptr;
+
+	printf("{");
+	while((p != NULL) && (i < size)) {
+		if((i % 8) == 0) {
+			printf("\n");
+			printf("    ");
+		}
+		printf("0x%02x, ", p[i]);
+		i++;
+	}
+	printf("\n};\n");
+    #else
+    (void) ptr;
+    (void) size;
+    #endif
+}
+
+static int last_val = 0x0f;
+
+static int check_result(unsigned char *data, int len)
+{
+	int i;
+	
+	for( ; len; ) {
+		last_val = (last_val + 1) % 16;
+		for(i = 0; i < 16; len--, i++, data++)
+			if(*data != last_val) {
+				return -1;
+			}
+	}
+    return 0;
+}
+
+static int r_offset;
+static int w_offset;
+
+static void init_offset()
+{
+    r_offset = 0;
+    w_offset = 0;
+}
+static void get_record(unsigned char *data, unsigned char *buf, int len)
+{
+    memcpy(buf, data+r_offset, len);
+    r_offset += len;
+}
+
+static void set_record(unsigned char *data, unsigned char *buf, int len)
+{
+    memcpy(data+w_offset, buf, len);
+    w_offset += len;
+}
+
+static void set_plain(unsigned char *plain, int rec)
+{
+    int i, j;
+    unsigned char *p = plain;
+
+    #define BLOCKSZ 16
+
+    for(i=0; i<(rec/BLOCKSZ); i++){
+        for(j=0; j<BLOCKSZ; j++)
+            *p++ = (i % 16);
+    }
+    //binary_dump(plain, rec);
```

Comment:
Should we remove this commented out line?

Reply:
fixed